ocaml/xapi/xapi_globs.ml

```diff
@@ -1017,6 +1017,14 @@ let conn_limit_unix = ref 1024
 
 let conn_limit_clientcert = ref 800
 
+let stunnel_cache_max_age = ref (180. *. 60.)
+
+let stunnel_cache_max_idle = ref (5. *. 60.)
+
+let coordinator_max_stunnel_cache = ref 70
+
+let member_max_stunnel_cache = ref 2
```

Comment:
This is very small and makes the stunnel cache potentially less effective. At the same time, it doesn't stop new connections. So is this worth doing now that the GC is made to run regularly?

Reply:
The timeout is 5 minutes though, which means that pool members can cause the coordinator to stop working for 5 minutes very easily during a burst of activity, even if that activity doesn't otherwise overlap with other hosts.

e.g. 
host1: 9 API calls concurrently at time t0
host2: 9 API calls concurently at time t1
...
hostN: 9 API calls concurently at time tN

if the distance between t0...tN is less than 5m then we'll have 9*64=576 open connections and begin causing failures either on the members or on the coordinator (depending on how the limits are set). And that happens even if we don't actually have 576 API calls at the same time, but just lets say 20 or 30 overall.

You are right that if we do actually have that many API calls going on in parallel then this doesn't help, because it is just a limit on idle stunnel connections: active stunnel connections can arbitrarily exceed this and still cause the coordinator to stop functioning.

A better approach would be to have an API proxy in front of the coordinator (written using Lwt or something else that doesn't need select) that turns synchronous calls into asynchronous calls to the real XAPI. That way it can handle any number of connections (within reason, the threads themselves will still need file descriptors to work, but e.g. SM would be very unlikely to encounter an error because all it does is make DB calls).

Also maybe we can do something with the pool SNI to direct pool internal API calls to another socket that could have another limit.